src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java

```diff
@@ -312,64 +306,92 @@ private Object getDefaultValue(String fieldName) {
 		return ret;
 	}
 
+	@Override
 	@SuppressWarnings("unchecked")
-	public <T> T getElementValue(String key) {
-		Object ret = this.elementValues.get(key);
-		if (ret == null) {
-			ret = getDefaultValue(key);
-		}
-		if (ret == null) {
-			ret = getReflectValue(key);
-		}
+	public <T extends CtExpression> T getValue(String key) {
+		return (T) getValueAsExpression(key);
+	}
 
+	private Object getValueAsObject(String key) {
+		CtExpression expr = getWrappedValue(key);
+		Object ret = convertElementToRuntimeObject(expr);
 		Class<?> type = getElementType(key);
+		return forceObjectToType(ret, type);
+	}
 
-		ret = this.convertValue(ret);
-
+	private Object forceObjectToType(Object ret, Class<?> type) {
 		if (type.isPrimitive()) {
 			if ((type == boolean.class) && (ret.getClass() != boolean.class)) {
-				ret = Boolean.parseBoolean(ret.toString());
+				return Boolean.parseBoolean(ret.toString());
 			} else if ((type == byte.class) && (ret.getClass() != byte.class)) {
-				ret = Byte.parseByte(ret.toString());
+				return Byte.parseByte(ret.toString());
 			} else if ((type == char.class) && (ret.getClass() != char.class)) {
-				ret = ret.toString().charAt(0);
+				return ret.toString().charAt(0);
 			} else if ((type == double.class) && (ret.getClass() != double.class)) {
-				ret = Double.parseDouble(ret.toString());
+				return Double.parseDouble(ret.toString());
 			} else if ((type == float.class) && (ret.getClass() != float.class)) {
-				ret = Float.parseFloat(ret.toString());
+				return Float.parseFloat(ret.toString());
 			} else if ((type == int.class) && (ret.getClass() != int.class)) {
-				ret = Integer.parseInt(ret.toString());
+				return Integer.parseInt(ret.toString());
 			} else if ((type == long.class) && (ret.getClass() != long.class)) {
-				ret = Long.parseLong(ret.toString());
+				return Long.parseLong(ret.toString());
 			} else if (type == short.class && ret.getClass() != short.class) {
-				ret = Short.parseShort(ret.toString());
+				return Short.parseShort(ret.toString());
 			}
 		}
-		if (type.isArray() && ret != null && ret.getClass() != type) {
```

Comment:
Are you sure with the removal of this piece of code? It seems related with the contract with annotation and arrays changed in https://github.com/INRIA/spoon/pull/1935

Reply:
the contract of #1935 is implemented at https://github.com/INRIA/spoon/pull/2034/files#diff-94d063d96db752ab6c7a094f53c75aceR395

do you see a missing test case to specify this?

Reply:
OK so you move the creation of the new array. It looks ok then. Merge?